itests/direct_data_onboard_test.go

```diff
@@ -102,6 +123,368 @@ func TestOnboardRawPiece(t *testing.T) {
+
+	allocations, err = client.StateGetAllocations(ctx, verifiedClientAddr, types.EmptyTSK)
+	require.NoError(t, err)
+	require.Len(t, allocations, 0)
+
+	evts, err := miner.FullNode.GetActorEvents(ctx, &types.ActorEventFilter{
+		MinEpoch: -1,
+		MaxEpoch: -1,
+	})
+	require.NoError(t, err)
+	for _, evt := range evts {
+		fmt.Printf("Got ActorEvent: %+v", evt)
+	}
+
+	eventsFromMessages := buildActorEventsFromMessages(t, ctx, miner.FullNode)
+	writeEventsToFile(t, ctx, miner.FullNode, eventsFromMessages)
+	for _, evt := range evts {
+		fmt.Printf("Got ActorEvent from messages: %+v", evt)
+	}
+
+	// TODO: compare GetActorEvents & SubscribeActorEvents & eventsFromMessages for equality
+}
+
+func buildActorEventsFromMessages(t *testing.T, ctx context.Context, node v1api.FullNode) []types.ActorEvent {
+	actorEvents := make([]types.ActorEvent, 0)
+
+	head, err := node.ChainHead(ctx)
+	require.NoError(t, err)
+	for height := 0; height < int(head.Height()); height++ {
+		// for each tipset
+		ts, err := node.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(height), types.EmptyTSK)
+		require.NoError(t, err)
+		for _, b := range ts.Blocks() {
+			// for each block
+			// alternative here is to go straight to receipts, but we need the message CID for our event
+			// list: node.ChainGetParentReceipts(ctx, b.Cid())
+			messages, err := node.ChainGetParentMessages(ctx, b.Cid())
+			require.NoError(t, err)
+			if len(messages) == 0 {
+				continue
+			}
+			for _, m := range messages {
+				receipt, err := node.StateSearchMsg(ctx, ts.Key(), m.Cid, -1, false)
+				require.NoError(t, err)
+				// receipt
+				if receipt.Receipt.EventsRoot != nil {
+					events, err := node.ChainGetEvents(ctx, *receipt.Receipt.EventsRoot)
+					require.NoError(t, err)
+					for _, evt := range events {
+						// for each event
+						addr, err := address.NewIDAddress(uint64(evt.Emitter))
+						require.NoError(t, err)
+						tsCid, err := ts.Key().Cid()
+						require.NoError(t, err)
+
+						actorEvents = append(actorEvents, types.ActorEvent{
+							Entries:     evt.Entries,
+							EmitterAddr: addr,
+							Reverted:    false,
+							Height:      abi.ChainEpoch(height),
+							TipSetKey:   tsCid,
+							MsgCid:      m.Cid,
+						})
+					}
+				}
+			}
+		}
+	}
+	return actorEvents
+}
+
+func writeEventsToFile(t *testing.T, ctx context.Context, node v1api.FullNode, events []types.ActorEvent) {
+	file, err := os.Create("block.out")
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, file.Close())
+	}()
+	write := func(s string) {
+		_, err := file.WriteString(s)
+		require.NoError(t, err)
+	}
+	claimKeyCbor, err := ipld.Encode(basicnode.NewString("claim"), dagcbor.Encode)
+	require.NoError(t, err)
+
+	for _, event := range events {
+		entryStrings := []string{
+			fmt.Sprintf("height=%d", event.Height),
+			fmt.Sprintf("msg=%s", event.MsgCid),
+			fmt.Sprintf("emitter=%s", event.EmitterAddr),
+			fmt.Sprintf("reverted=%t", event.Reverted),
```

Comment:
This information is not persisted on message receipts. How will client be informed about reverted events if they only depend on message receipts ?